nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/MaxPoolWithArgmax.java

```diff
@@ -0,0 +1,34 @@
+package org.nd4j.linalg.api.ops.custom;
+
+import lombok.NonNull;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+
+public class MaxPoolWithArgmax extends DynamicCustomOp {
```

Comment:
This op is missing all the args and configuration.
It should basically be the same as MaxPool2d op
Take a look at these:
https://www.tensorflow.org/api_docs/python/tf/nn/max_pool_with_argmax
https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/pooling/maxpool2d.cpp
https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/pooling/maxpool_with_argmax.cpp

Note that the libnd4j format iargs are the same for maxpool2d and maxpool_with_argmax